CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}")
 set(CMAKE_BINARY_DIR "${CMAKE_SOURCE_DIR}/bin")
-set(CMAKE_BUILD_TYPE "RelWithDebInfo")
+# set(CMAKE_BUILD_TYPE "RelWithDebInfo")
```

Comment:
@johannct I noticed that this was hardcoded here. I've only commented it out because it was interfering with the conditional logic on line 69. I'm happy to revert this change if needed.

Reply:
Can you expand on the overall take on conditional logic? RelWithDebInfo seems to be a standard for preformance builds but still with debug symbols. Perhaps we should lay down the different builds we need/want?

Reply:
Initially I wanted use `CMAKE_BUILD_TYPE="Debug"` be the trigger to include the coverage flags when building, but I'm happy to go back to the default that was here, and we'll look for an env var to conditionally include the coverage flags.